orsopy/fileio/data_source.py

```diff
@@ -61,30 +64,31 @@ class Sample(Header):
     name: str
     category: Optional[str] = None
     composition: Optional[str] = None
-    description: Optional[Union[list, str]] = None
-    environment: Optional[Union[List[str], str]] = None
+    description: Optional[str] = None
+    environment: Optional[List[str]] = None
     sample_parameters: Optional[Dict] = field(
         default=None, metadata={"description": "Using keys for parameters and Value* objects for values."}
     )
 
 
-# Enum does not work with yaml, if we really want this it has to be handle
-# as special case.
-#
-# class Polarization(str, enum.Enum):
-#     """
-#     The first symbol indicates the magnetisation direction of the incident
-#     beam. An optional second symbol indicates the direction of the scattered
-#     beam, if a spin analyser is present.
-#     """
-#
-#     unpolarized = "unpolarized"
-#     p = "+"
-#     m = "-"
-#     mm = "--"
-#     mp = "-+"
-#     pm = "+-"
-#     pp = "++"
+class Polarization(str, Enum):
+    """
+    The first symbol indicates the magnetisation direction of the incident
+    beam. An optional second symbol indicates the direction of the scattered
+    beam, if a spin analyser is present.
+    """
+
+    unpolarized = "unpolarized"
+    p = "p"
+    m = "m"
+    mm = "mm"
+    mp = "mp"
+    pm = "pm"
+    pp = "pp"
```

Comment:
There seemed to be general agreement on Slack that we could add ```["?m", "m?", "?p", "p?"]``` to represent half-polarized states (where the part that has unknown/unmeasured polarization is then explicit) - these would replace the bare ```l"p", "m"]``` that don't provide information about which half is unknown.

Reply:
Question marks would make it problematic to use as attribute names. Jochen and I preferred to go with "o" instead, hope that's OK.

Reply:
why would they be used as attribute names, instead of values?